scripts/perftests/PerformanceTestUtilities.ps1

```diff
@@ -0,0 +1,214 @@
+# Contains all the utility methods used by the performance tests.
+
+    # Downloads a nuget exe in the given directory.
+    # The exe will be found at $directory/$version/nuget.exe
+    function DownloadNuGetExe([string]$version, [string]$downloadDirectory)
+    {        
+        $NuGetExeUriRoot = "https://dist.nuget.org/win-x86-commandline/v"
+        $NuGetExeSuffix = "/nuget.exe"
+
+        $url = $NuGetExeUriRoot + $version + $NuGetExeSuffix
+        $Path =  $downloadDirectory + "\" + $version
+        $ExePath = $Path + "\nuget.exe"
+
+        if (!(Test-Path($ExePath)))
+        {
+            Log "Downloading $url to $ExePath" -color "Green"
+            New-Item -ItemType Directory -Force -Path $Path > $null
+            Invoke-WebRequest -Uri $url -OutFile $ExePath
+        }
+        return GetAbsolutePath $ExePath
+    }
+
+    
+    # The format of the URL is assumed to be https://github.com/NuGet/NuGet.Client.git. The result would be NuGet-Client-git
+    function GenerateNameFromGitUrl([string]$gitUrl)
+    {
+        return $gitUrl.Substring($($gitUrl.LastIndexOf('/') + 1)).Replace('.','-')
+    }
+
+    # Appends the log time in front of the log statement with the color specified. 
+    function Log([string]$logStatement, [string]$color)
+    {
+        if(-not ([string]::IsNullOrEmpty($color)))
```

Comment:
Invert the logic for improved readability.  Because this:

    if (condition) { }
    else { }  // not condition

...is more readable than an implicit double negative:

    if (not condition) { }
    else { } // not (not condition)